openpype/hosts/maya/plugins/publish/convert_gltf_shader.py

```diff
@@ -0,0 +1,78 @@
+import os
+from maya import cmds
+import pyblish.api
+
+from openpype.pipeline import publish
+
+
+class ConvertGLSLShader(publish.Extractor):
+    """
+    Converting StingrayPBS material to GLSL Shaders
+    specially for the glb export through Maya2GLTF plugin
+
+    """
```

Comment:
I personally think it's weird this is changing the local work scene. A plug-in shouldn't do that.
A validator could do that potentially through a repair method if needed.

I also feel that this plug-in on family `gltf` might be too specific to a certain workflow. I've done about 4-5 gltf projects using this maya2gltf plug-in and I've never had the need to use `StingrayPBR` so far. Should this be a `gltf.unreal` or whatever subfamily? 

I'm not sure what the naming conventions are based on, etc. but it might be good in the docstring to mention and clarify that. Is it unreal? Is it something else?

Reply:
The purpose of this validator is to check the textures with the naming standard of ue5 style. We can set this validator as optional and allows users to enable if they want it here, similar to the gltf texture extractor

Reply:
Publishing should be _idempotent_. Publishing again should do the exact same processing. Because this plugin changes the workfile that's not the case at all.